dt-workflows/workflows-execution-handler.php

```diff
@@ -583,14 +583,87 @@ private static function action_append( $field_type, $field_id, $value ): array {
+                    if ( $obj['type'] == 'connection' || $obj['type'] == 'user_select' ) {
+                        return true;
+                    }
+                    return false;
+                } );
+
+                // loop through each field, checking if the comment attempts to reference one of the fields
+                foreach ( $fields as $field => $data ) {
+                    if ( str_contains( $value, '{' . $field . '}' ) ) {
+                        $replacement = '';
+                        $current = 0;
+                        $references = $post[$field]; // get array of references from the field
+
+                        // special case: if the field is 'assigned_to', it is not an array of references,
+                        // but an array of the pieces of the 'assigned_to' field
+                        if ( $field == 'assigned_to' ) {
+                            $replacement = $replacement . '@[' . $references['display'] . '](' . $references['id'] . ')';
+                        } else {
+                            // otherwise loop through references, building replacement string
+                            foreach ( $references as $reference ) {
+                                ++$current;
+
+                                $id = '';
+                                $reference_post = DT_Posts::get_post( $data['post_type'], $reference['ID'], false, false );
+                                // if the post corresponds to a user, @mention that user, otherwise just the post name
+                                if ( !empty( $reference_post['corresponds_to_user'] ) ) {
+                                    $id = $reference_post['corresponds_to_user'];
+                                    $replacement = $replacement . '@[' . $reference['post_title'] . '](' . $id . ')';
+                                } else {
+                                    $replacement = $replacement . $reference['post_title'];
+                                }
+                                // if not the last reference, add a comma & space
+                                if ( $current != sizeof( $references ) ) {
+                                    $replacement = $replacement . ', ';
+                                }
+                            }
+                        }
+
+                        // replace the {field} in the comment w/ the references
+                        $value = str_replace( '{' . $field . '}', $replacement, $value );
+                    }
+                }
+
+                // create the comment
                 $updated['notes'] = [
                     $value,
                 ];
+
         }
 
         return $updated;
     }
 
+    private static function get_references( $field_name, $post ) {
+        $replacement = '';
+        $current = 0;
+        $references = $post[$field_name];
+        foreach ( $references as $reference ) {
+            ++$current;
```

Comment:
Instead of manually tracking the array index, you can get that using this syntax:

```
foreach ( $references as $current => $reference)
```
See https://www.php.net/manual/en/control-structures.foreach.php